### PR TITLE
Create createBlockSyner() in `chainservice.go` and clean some comments 

### PR DESCRIPTION
**Pull request by Liuhaai**
move the creation of `BlockSyner` to a subfunc

### CODE REVIEW

File: pkg/routine/recurringtask.go [hunk 1]
```diff
@@ -46,32 +46,27 @@ func NewRecurringTask(t Task, i time.Duration, ops ...RecurringTaskOption) *Recu
...
-			case <-t.ch:
+			case <-t.done:
 				return
 			case <-t.ticker.C:
 				t.t()
 			}
 		}
 	}()
-
-	<-ready
```
Comment (dustinxie) [inline pkg/routine/recurringtask.go:65]: this is needed to make sure the go routine is active before `Start()` exits

File: pkg/routine/recurringtask.go [hunk 2]
```diff
@@ -46,32 +46,27 @@ func NewRecurringTask(t Task, i time.Duration, ops ...RecurringTaskOption) *Recu
 }
 
 // Start starts the timer
-func (t *RecurringTask) Start(ctx context.Context) error {
+func (t *RecurringTask) Start(_ context.Context) error {
 	t.ticker = t.clock.Ticker(t.interval)
-	ready := make(chan struct{})
 	go func() {
-		close(ready)
```
Comment (dustinxie) [inline pkg/routine/recurringtask.go:53]: needed, see below

**Comment by CoderZhi** [comment]
one topic in one pr

**Comment by Liuhaai** [comment]
> one topic in one pr

ok

**Pull request closed without merging**